src/plugins/intel_gpu/src/runtime/memory_pool.cpp

```diff
@@ -309,25 +412,152 @@ void memory_pool::clear_pool_for_network(uint32_t network_id) {
+    if (type == allocation_type::usm_host) {
+        return host_mem_size;
+    } else {
+        return (total_mem_size - host_mem_size);
+    }
+}
+
+void memory_pool::dump(uint32_t net_id, uint32_t iter, std::string dump_dir_path) {
+    dump_to_screen(net_id, iter);
+    if (!dump_dir_path.empty())
+        dump_to_file(net_id, iter, dump_dir_path);
+}
+
+void memory_pool::dump_to_file(uint32_t net_id, uint32_t iter, std::string dump_dir_path) {
+    const std::string dump_file_name = "dump_runtime_memory_pool_net_" + std::to_string(net_id) + "_iter_" + std::to_string(iter) + ".csv";
+    const std::string desc = "pool_type,layout,mem_ptr,mem_type,mem_pool_size,prim_id,unique_id,mem_size";
+    const std::string dump_path = dump_dir_path + dump_file_name;
+    std::ofstream of(dump_path);
+    if (of.is_open()) {
+        of << desc << std::endl;
+        for (auto mem : _non_padded_pool) {
+            for (auto user : mem.second._users) {
+                of << "non_padded_pool,," << mem.second._memory->buffer_ptr() << "," << mem.second._type << ","
+                    << mem.first << "," << user._prim_id << "," << user._unique_id << "," << user._mem_size << std::endl;
+            }
         }
+
+        for (auto mem : _padded_pool) {
+            for (auto record : mem.second) {
+                const size_t mem_pool_size = record._memory->size();
+                for (auto user : record._users) {
+                    of << "padded_pool," << mem.first.to_short_string() << "," << record._memory->buffer_ptr() << "," << record._type << ","
+                        << mem_pool_size << "," << user._prim_id << "," << user._unique_id << "," << user._mem_size << std::endl;
+                }
+            }
+        }
+        for (auto mem : _no_reusable_pool) {
+            for (auto user : mem.second._users) {
+                of << "no_reused_pool,," << mem.second._memory->buffer_ptr() << "," << mem.second._type << ","
+                    << mem.first << "," << user._prim_id << "," << user._unique_id << "," << user._mem_size << std::endl;
+            }
+        }
+        std::cout << "Dump file to " << dump_path << std::endl;
     }
-    GPU_DEBUG_COUT << "========== padded pool (" << _padded_pool.size() << " records) ==========" << std::endl;
-    for (auto mem : _padded_pool) {
-        GPU_DEBUG_COUT << " layout: " << mem.first.to_short_string() << std::endl;
-        for (auto record : mem.second) {
-            GPU_DEBUG_COUT << "    " << record._memory->buffer_ptr() << ", type: " << record._type << ", users : " << std::endl;
-            for (auto user : record._users) {
-                GPU_DEBUG_COUT << "    --- " << user._prim_id << " (" << user._unique_id << std::endl;
+}
+
+void memory_pool::dump_to_screen(uint32_t net_id, uint32_t iter) {
+    GPU_DEBUG_COUT << "Dump memory pool of network (net_id : " << net_id << ", iter : " << iter << ")" << std::endl;
```

Comment:
Could you fix these debug code only to be included when the DEBUG_CAP is ON?

Reply:
Done